src/dotnetcore31/AzureFunctions.Durable.Chaining/GitHub/GitHubInfoOrchestrator.cs

```diff
@@ -0,0 +1,68 @@
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Octokit;
+
+namespace Durable.Chaining.GitHub
+{
+    public static class GitHubInfoOrchestrator
+    {
+        [FunctionName("GitHubInfoOrchestrator")]
+        public static async Task<string> RunOrchestrator(
+            [OrchestrationTrigger] IDurableOrchestrationContext context)
+        {
+            var input = context.GetInput<string>();
+
+            var userName = await context.CallActivityAsync<string>("GitHubInfoOrchestrator_GetRepositoryDetailsByName", input);
+
+            var userDetails = await context.CallActivityAsync<string>("GitHubInfoOrchestrator_GetUserDetailsByName", userName);
+
+            return userDetails;
+        }
+
+        [FunctionName("GitHubInfoOrchestrator_GetRepositoryDetailsByName")]
+        public static async Task<string> GetRepositoryDetails([ActivityTrigger] string name, ILogger log)
+        {
+            var githubClient = new GitHubClient(new ProductHeaderValue("azure-functions-university"));
+
+            var request = new SearchRepositoriesRequest(name);
+
+            var searchResult = await githubClient.Search.SearchRepo(request);
+            var repository = searchResult.Items.Single(x => x.Name == name);
+
+            return repository.Owner.Login;
+        }
+
+        [FunctionName("GitHubInfoOrchestrator_GetUserDetailsByName")]
+        public static async Task<string> GetUserDetailsByName([ActivityTrigger] string name, ILogger log)
+        {
+            var githubClient = new GitHubClient(new ProductHeaderValue("azure-functions-university"));
+
+            var user = await githubClient.User.Get(name);
+
+            var serializedUser = JsonSerializer.Serialize<object>(user);
+            return serializedUser;
```

Comment:
Is this serialization neccesary? the githubClient.User.Get returns a `User` object. Can this object be returned from this function without manual serialization? Ideally I'd like to prevent addiitonal serialization steps since Durable Functions will also do serialization when state is persisted to the storage account.

Reply:
Yes it is needed because the default serializer (`System.Text.Json`) do not serialize base classes properties by default. To force the serliazation of all of the properties, you need to serialize the object as `object` type - sorry for word play - or special attributes that the `Octokit`'s classes don't use.

You can change the default behaviour by changing default options at startup, but I don't think this is the case.

More info [here](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/polymorphism?pivots=dotnet-6-0#serialize-properties-of-derived-classes)

Reply:
I got it, thanks for the explanation!